components/datetime/src/fields/length.rs

```diff
@@ -15,10 +15,8 @@ pub enum LengthError {
 impl std::error::Error for LengthError {}
 
 #[derive(Debug, Eq, PartialEq, Clone, Copy, Ord, PartialOrd)]
-#[cfg_attr(
-    feature = "provider_serde",
-    derive(serde::Serialize, serde::Deserialize)
-)]
+#[cfg_attr(feature = "datagen", derive(serde::Serialize))]
+#[cfg_attr(feature = "serialize", derive(serde::Deserialize))]
```

Comment:
Thought: These are library types. I assume they were guarded on `provider_serde` because that's the only thing in ICU4X that needed the serde impls, but, they should probably actually be under the normal `serde` (`serialize`) feature. I think maybe make the distinction only for proper data structs?

Reply:
Makes sense. By and large I was conservative in what gets treated as a library type for this logic, with the idea that it's easier to be consistent in the large PR and make small changes later. I can push up some fixes.